pkg/sqlite/load.go

```diff
@@ -192,86 +199,101 @@ func (s *SQLLoader) AddPackageChannels(manifest registry.PackageManifest) error
 			if err != nil {
-				return err
+				errs = append(errs, err)
+				break
 			}
 
 			if err := s.addProvidedAPIs(tx, channelEntryCSV, currentID); err != nil {
-				return err
+				errs = append(errs, err)
 			}
 
 			skips, err := channelEntryCSV.GetSkips()
 			if err != nil {
-				return err
+				errs = append(errs, err)
 			}
 
 			for _, skip := range skips {
 				// add dummy channel entry for the skipped version
 				skippedChannelEntry, err := addChannelEntry.Exec(c.Name, manifest.PackageName, skip, depth)
 				if err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				skippedID, err := skippedChannelEntry.LastInsertId()
 				if err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				// add another channel entry for the parent, which replaces the skipped
 				synthesizedChannelEntry, err := addChannelEntry.Exec(c.Name, manifest.PackageName, channelEntryCSVName, depth)
 				if err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				synthesizedID, err := synthesizedChannelEntry.LastInsertId()
 				if err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				if _, err = addReplaces.Exec(skippedID, synthesizedID); err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
 				if err := s.addProvidedAPIs(tx, channelEntryCSV, synthesizedID); err != nil {
-					return err
+					errs = append(errs, err)
+					continue
 				}
 
-				depth += 1
+				depth++
 			}
 
 			// create real replacement chain
 			replaces, err := channelEntryCSV.GetReplaces()
 			if err != nil {
-				return err
+				errs = append(errs, err)
+				break
 			}
 
 			if replaces == "" {
 				// we've walked the channel until there was no replacement
 				break
 			}
 
-			replaced, err := s.getCSV(tx, replaces)
+			replacedChannelEntry, err := addChannelEntry.Exec(c.Name, manifest.PackageName, replaces, depth)
 			if err != nil {
-				return fmt.Errorf("%s specifies replacement that couldn't be found", c.CurrentCSVName)
-			}
-
-			replacedChannelEntry, err := addChannelEntry.Exec(c.Name, manifest.PackageName, replaced.GetName(), depth)
-			if err != nil {
-				return err
+				errs = append(errs, err)
+				break
 			}
 			replacedID, err := replacedChannelEntry.LastInsertId()
 			if err != nil {
-				return err
+				errs = append(errs, err)
+				break
 			}
-			_, err = addReplaces.Exec(replacedID, currentID)
-			if err != nil {
-				return err
+			if _, err = addReplaces.Exec(replacedID, currentID); err != nil {
+				errs = append(errs, err)
+				break
+			}
+			if _, err := s.getCSV(tx, replaces); err != nil {
+				errs = append(errs, fmt.Errorf("%s specifies replacement that couldn't be found", c.CurrentCSVName))
+				break
 			}
+
 			currentID = replacedID
 			channelEntryCSVName = replaces
-			depth += 1
+			depth++
 		}
 	}
-	return tx.Commit()
+
+	if err := tx.Commit(); err != nil {
```

Comment:
We should revisit this, and probably have more granular transactions. This one failing would nullify a lot of the work above to load as much as possible.